File: src/main/java/org/jukeboxmc/form/FormResponse.java
package org.jukeboxmc.form;

import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * @author GoMint
 * @version 1.0
 */
public class FormResponse {

    private final Map<String, Object> answers = new HashMap<>();

    public void addAnswer( String id, Object data ) {
        this.answers.put( id, data );
    }

    public @Nullable Boolean getToggle(String id ) {
        Object val = this.answers.get( id );
        if ( val != null ) {
            if ( val instanceof Boolean ) {
                return (Boolean) val;
            }
        }

        return null;
    }

    public @Nullable String getStepSlider(String id ) {
        Object val = this.answers.get( id );
        if ( val != null ) {
            if ( val instanceof String ) {
                return (String) val;
            }
        }

        return null;
    }

    public @Nullable Float getSlider(String id ) {
        Object val = this.answers.get( id );
        if ( val != null ) {
            if ( val instanceof Double ) {
                return ( (Double) val ).floatValue();
            }
        }

        return null;
    }

    public @Nullable String getInput(String id ) {
        Object val = this.answers.get( id );
        if ( val != null ) {
            if ( val instanceof String ) {
                return (String) val;
            }
        }

        return null;
    }

    public @Nullable String getDropbox(String id ) {
        Object val = this.answers.get( id );
        if ( val != null ) {
            if ( val instanceof String ) {
                return (String) val;
            }
        }

        return null;
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/FormResponse.kt
package org.jukeboxmc.api.form

/**
 * @author Kaooot
 * @version 1.0
 */
class FormResponse {

    private val answers = hashMapOf<String, Any?>()

    fun addAnswer(id: String, data: Any?) {
        this.answers[id] = data
    }

    fun getToggle(id: String): Boolean? {
        val obj = this.answers[id]
        if (obj != null && obj is Boolean) {
            return obj
        }
        return null
    }

    fun getStepSlider(id: String): String? {
        val obj = this.answers[id]
        if (obj != null && obj is String) {
            return obj
        }
        return null
    }

    fun getSlider(id: String): Float? {
        val obj = this.answers[id]
        if (obj != null && obj is Double) {
            return obj.toFloat()
        }
        return null
    }

    fun getInput(id: String): String? {
        val obj = this.answers[id]
        if (obj != null && obj is String) {
            return obj
        }
        return null
    }

    fun getDropbox(id: String): String? {
        val obj = this.answers[id]
        if (obj != null && obj is String) {
            return obj
        }
        return null
    }
}